### PR TITLE
Active drive

### PR DESCRIPTION
**Pull request by kvanhijf**

### CODE REVIEW

File: source/asdmanager.py [hunk 1]
```diff
@@ -52,7 +54,7 @@ def setup():
         sys.exit(1)
 
     ipaddresses = check_output("ip a | grep 'inet ' | sed 's/\s\s*/ /g' | cut -d ' ' -f 3 | cut -d '/' -f 1", shell=True).strip().splitlines()
```
Comment (JeffreyDevloo) [inline source/asdmanager.py]: Identical to my review on FWK. Move this to extensions

File: source/controllers/asd.py [hunk 2]
```diff
@@ -52,6 +52,14 @@ def create_asd(disk):
         if disk.state == 'MISSING':
             raise RuntimeError('Cannot create an ASD on missing disk {0}'.format(disk.name))
 
+        ipaddresses = Configuration.get('{0}/network|ips'.format(ASDController.CONFIG_ROOT))
+        if len(ipaddresses) == 0:
+            ipaddresses = ASDController._local_client.run("ip a | grep 'inet ' | sed 's/\s\s*/ /g' | cut -d ' ' -f 3 | cut -d '/' -f 1", allow_insecure=True).strip().splitlines()
```
Comment (JeffreyDevloo) [inline source/controllers/asd.py]: Move to extensions.

File: source/controllers/asd.py [hunk 3]
```diff
@@ -52,6 +52,14 @@ def create_asd(disk):
...
+            ipaddresses = [found_ip.strip() for found_ip in ipaddresses if not found_ip.strip().startswith('127.')]
+            if len(ipaddresses) == 0:
+                raise RuntimeError('Could not find any IP on the local node')
+            ipaddresses = [ipaddresses[0]]  # We make list of first IP found on system if None were configured during initial setup
```
Comment (JeffreyDevloo) [inline source/controllers/asd.py]: Currently a WIP to register with all ips. For the initial version this is fine but it has to be adjusted for the final one.

**Comment by kvanhijf** [comment]
https://github.com/openvstorage/framework/pull/1702 --> openvstorage-2.10.0-dev.1502181021.4906607
https://github.com/openvstorage/iscsi-manager/pull/17 --> iscsi-manager-0.2.0-dev.1502180042.cdcc95c
https://github.com/openvstorage/alba-asdmanager/pull/260 --> openvstorage-sdm-1.10.0-dev.1502180138.57073dd
https://github.com/openvstorage/framework-extensions/pull/18 --> openvstorage-extensions-0.2.0-dev.1502180232.b170e86
https://github.com/openvstorage/framework-iscsi-plugin/pull/14 --> openvstorage-iscsi-plugin-0.2.0-dev.1502179954.bf76108
https://github.com/openvstorage/framework-alba-plugin/pull/432 --> openvstorage-backend-1.10.0-dev.1502180753.6e1c736

**Pull request merged**